apps/server/lib/http/auth.ts

```diff
@@ -0,0 +1,75 @@
+import {
+	JellyfishKernel,
+	SessionContract,
+} from '@balena/jellyfish-types/build/core';
+import bcrypt from 'bcrypt';
+import _ from 'lodash';
+
+// TODO: Move this definition elsewhere, and inject it into the middleware.
+const sessionIdAndTokenValidator =
+	(context, jellyfish: JellyfishKernel) =>
+	async (session: string, sessionToken?: string): Promise<boolean> => {
+		try {
+			const card = await jellyfish.getCardById<SessionContract>(
+				context,
+				jellyfish.sessions!.admin,
+				session,
+			);
+			return bcrypt.compare(sessionToken, card!.data.token!.authentication);
+		} catch (e) {
+			return false;
+		}
+	};
+
+// TODO: Move this definition elsewhere, and inject it into the middleware.
+const sessionIdValidator =
```

Comment:
I don't think this function is necessary. `sessionIdAndTokenValidator()` also queries the database for the session contract. We could return `false` if the card doesn't exist at that point.